### PR TITLE
Add AK/SK Management design proposal

### PR DESCRIPTION
**Pull request by himanshuvar**
Adding design specification for the Multi-Cloud AK/SK Management in OpenSDS.

### CODE REVIEW

**Comment by leonwanghui** [comment]
@himanshuvar Please move this proposal draft under `capri` folder.

**Comment by himanshuvar** [comment]
> @himanshuvar Please move this proposal draft under `capri` folder.

Done.

File: specs/capri/MultiCloud_AK-SK_Management_Design.md [hunk 1]
```diff
@@ -0,0 +1,74 @@
...
+### Signature Calculation and Validation
+
+API requests to multi-cloud containing the authentication information must include a signature which is computed based on the request elements. To calculate a signature, concatenate request elements to form a string (string to sign) and then create a signing key using secret access key. Signing key is derived from the credential scope, which means that there is no need to include the key in the request. Then signing key is used to calculate the hash-based message authentication code (HMAC) of the string to sign.
+When API handler receives the request, it performs the same steps to calculate the signature included in the request. API handler then compares its calculated signature to the one sent with the request. If the signatures match, the request is processed. If the signatures don't match, the request is denied.
+
+AWS Signature Version 4 signing library i.e. aws-sigv4 can be used to generate the signature.
+
+Following is the illustration for the signature calculation and validation process.
+
+![Multi-Cloud Signature Calculation Diagram](multicloud_signature_calculation.PNG?raw=true "Multi-Cloud Signature Calculation Diagram")
```
Comment (xing-yang) [inline specs/capri/MultiCloud_AK-SK_Management_Design.md]: As discussed in the OpenSDS technical meeting, can you please make the text in this calculation diagram bigger so people can read it?

Comment (himanshuvar) [reply specs/capri/MultiCloud_AK-SK_Management_Design.md]: Done

**Comment by wisererik** [comment]
@himanshuvar could you please add the api impact.  Which apis do need to be authenticated? What's the change of API requests?

**Comment by xing-yang** [comment]
lgtm

**Pull request merged**